pkl-codegen-java/src/main/kotlin/org/pkl/codegen/java/JavaCodeGenerator.kt

```diff
@@ -541,8 +541,8 @@ class JavaCodeGenerator(
       val builder =
         TypeSpec.classBuilder(javaPoetClassName.simpleName()).addModifiers(Modifier.PUBLIC)
 
-      // stateless classes aren't instantiable by choice, i.e., no public ctor is generated
-      val isInstantiable = !pClass.isAbstract && allProperties.isNotEmpty()
+      // stateless module classes are non-instantiable by choice
+      val isInstantiable = !pClass.isAbstract && !(isModuleClass && allProperties.isEmpty())
```

Comment:
We should let empty modules to be instantiable too, unless I'm missing something obvious here?

```suggestion
      val isInstantiable = !pClass.isAbstract;
```

Reply:
I just narrowed the exception from "stateless module classes" to "stateless final module classes".
This is consistent with codegen-kotlin, which generates object declarations in such cases.
I guess this is trying to generate nicer code for the "I'm a pure schema" case.

Reply:
I don't know if modules are any different here.

This module should be able to be mapped to Java, for example:

```pkl
module com.example.Foo
// I'm totally empty!
```

Reply:
With this PR, both code generators should behave exactly the same.
A future PR could remove the special handling of stateless final module classes.
This will generate some top-level code (equals/hashCode/toString/with) that sits next to (possibly above) the class declarations.

Reply:
Okay, fair enough!